### PR TITLE
F/local forage

### PR DESCRIPTION
**Pull request by hiqbal01**

### CODE REVIEW

File: src/services/MetaService.ts [hunk 1]
```diff
@@ -68,13 +78,15 @@ export class MetaService {
...
   async get(requested: string[], layout?: string): Promise<FieldMap[]> {
+    await this.initialized; // This is ensuring that our initialization is complete
     const missing = this.missing(requested);
 
     if (missing.length || layout) {
       this.parameters.fields = missing.join(',');
       if (layout) {
         this.parameters.layout = layout;
       }
+      console.log('Making meta call for these fields', requested.toString(), this.endpoint);
```
Comment (monroepe) [inline src/services/MetaService.ts]: Do we need this console.log? I prefer not to have any console logs.

File: src/services/MetaService.ts [hunk 2]
```diff
@@ -85,7 +97,32 @@ export class MetaService {
     return this.extract(requested);
   }
 
+  /**
+   * Make http request to get track data. Response data will be parsed, then the Promise will be resolved.
+   */
+  async getTracks(): Promise<BullhornTrack[]> {
+    await this.initialized; // This is ensuring that our initialization is complete
```
Comment (monroepe) [inline src/services/MetaService.ts]: Also not a big deal one way or another. But I don't think you really need the comment here. I think it's pretty clear with the naming of this.initialized.

File: src/services/MetaService.ts (hunk 1 above)
Comment (hiqbal01) [reply src/services/MetaService.ts]: yeah I'll take this out

File: src/services/MetaService.ts (hunk 2 above)
Comment (hiqbal01) [reply src/services/MetaService.ts]: I wanted to leave that in there so future developers can see that they need to make sure this is resolved before they do other things. If you think we don't need it, I can take it out.

Comment (hiqbal01) [reply src/services/MetaService.ts]: Taking it out

Comment (monroepe) [reply src/services/MetaService.ts]: Ah that makes sense. No it's totally fine. I will resolve this comment.

File: src/utils/Cache.ts [hunk 3]
```diff
@@ -99,29 +139,85 @@ export class Cache {
...
+  // Angular patches fetch in Safari and the test to check if fetch is native code returns false
+  // Once LocalForage fixes this bug, we can remove this method
+  // tslint:disable-next-line
+  localforage['originalSupports'] = localforage.supports;
+  localforage.supports = (driver: string) => {
+    try {
+      const isSafari = /(Safari|iPhone|iPad|iPod)/.test(navigator.userAgent) &&
+            !/Chrome/.test(navigator.userAgent) &&
+            !/BlackBerry/.test(navigator.platform);
+      if (isSafari) {
```
Comment (amrutha-rajiv) [inline src/utils/Cache.ts]: there's another class in taurus that's checking browser name, it just doesn't check the version.

**Pull request merged**